### ISSUE TITLE
cmd/compile: validity of program depends on method declaration order

### ISSUE THREAD

**Issue by griesemer** [golang/go#23202]
https://play.golang.org/p/WCktUidLyc_ is accepted while the equivalent program https://play.golang.org/p/Z-B9rBhYILd fails with an error. The only difference is the order of the method declarations.

Esoteric case; recording this so we're aware of it.

**Comment by bradfitz** [comment]
Hey @andybons, that first snippet doesn't load.

**Comment by andybons** [comment]
Yes it does. The trailing underscore gets stripped with some URL parsers (like in Gmail). Make sure it’s there and it loads for me.

**Comment by bradfitz** [comment]
Maybe we shouldn't generate snippets with trailing underscores then.

**Comment by ianlancetaylor** [comment]
(Seems to work with gccgo.)

**Comment by andybons** [comment]
Pushed new version of play and updated original comment with link that is properly linkified by GitHub. Sorry to derail this bug, Robert.
